src/test/suite/compile.test.ts

```diff
@@ -0,0 +1,106 @@
+import * as assert from 'assert';
+import { after, before, describe, it, beforeEach } from 'mocha';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+import { Uri, window, commands, workspace, TextDocument, TextEditor, Position, SnippetString } from 'vscode';
+
+const logPath: string = '/tmp/vscode-inmanta.log';
+const workspaceUri: Uri = Uri.file(path.resolve(__dirname, '../../../src/test/workspace'));
+const libsPath: string = path.resolve(workspaceUri.fsPath, 'libs');
+const envPath: string = path.resolve(workspaceUri.fsPath, '.env');
+
+const modelUri: Uri = Uri.file(path.resolve(workspaceUri.fsPath, 'main.cf'));
+
+function waitForCompile(timeout: number): Promise<boolean> {
+	const start = Date.now();
+	return new Promise<boolean>((resolve, reject) => {
+		const readLogInterval = setInterval(() => {
+			if (Date.now() - start > timeout) {
+				reject(new Error("Timeout reached"));
+			} else {
+				fs.ensureFileSync(logPath);
+				fs.readFile(logPath, 'utf-8', (err, data) => {
+					if (err) {
+						console.log(err);
```

Comment:
Call reject? It's been a while since I've worked with javascript so I'm a bit out of sync.

Reply:
Not sure about that, if for some reason the read operation fails, it doesn't mean it couldn't succeed later on (before the timeout expire).  Like for example if the file doesn't exist yet, or is temporarily unreachable.

Reply:
Right